src/builder.rs

```diff
@@ -568,11 +568,28 @@ impl<'a, 'gcc, 'tcx> BuilderMethods<'a, 'tcx> for Builder<'a, 'gcc, 'tcx> {
     ) {
         let mut gcc_cases = vec![];
         let typ = self.val_ty(value);
-        for (on_val, dest) in cases {
-            let on_val = self.const_uint_big(typ, on_val);
-            gcc_cases.push(self.context.new_case(on_val, on_val, dest));
+        // FIXME: This is a workaround for a libgccjit limitation.
+        // Currently, libgccjit can't directly create 128 bit intigers.
+        // Since switch cases must be values, and casts are not constant, we can't use 128 bit switch cases.
+        // In such a case, we will simply fall back to an if-ladder.
+        // This *may* be slower than a native switch, but a slow working solution is better than none at all.
+        if typ.is_i128(self) || typ.is_u128(self) {
+            for (on_val, dest) in cases {
+                let on_val = self.const_uint_big(typ, on_val);
+                let is_case =
+                    self.context.new_comparison(None, ComparisonOp::Equals, value, on_val);
+                let next_block = self.current_func.borrow().unwrap().new_block("");
```

Comment:
I would give the block a dummy name just to make it clearer in the generated IR.
(We can also use the method `current_func()` to make this simpler.)
```suggestion
                let next_block = self.current_func().new_block("case");
```